docs/getting-started/v3-with-cash-app-pay.md

```diff
@@ -0,0 +1,328 @@
+                }
+               ... // other states 
+            }
+        }
+    }
+
+```
+
+
+## Step 7: Authorize when customer clicks button
+
+When beginning the authorization request, Cash App Pay SDK will redirect the Customer to Cash App, where they can approve or decline the grant request.
+
+```kotlin
+bindings.cashappPayButton.apply {
+    setOnClickListener { _ ->
+        cashAppPay.authorizeCustomerRequest()
+    }
+}
+```
+
+## Step 8: Disable button when authorization is in flight
+
+Avoid duplicate button clicks by disabling once an authorization is detected.
+
+```kotlin
+private val cashAppPayListener =
+    object : CashAppPayListener {
+        override fun cashAppPayStateDidChange(newState: CashAppPayState) {
+            when (newState) {
+                Authorizing -> {
+                    bindings.cashappPayButton.isEnabled = false
+                }
+               ... // other states 
+            }
+        }
+    }
+```
+
+
+## Step 9: Respond to Approved or Declined
+
+Cash App Pay SDK will redirect back to your app (per the supplied redirect URI). `CashAppPayListener` will receive either a `Approved` or `Declined` response.
+
+```kotlin
+private val cashAppPayListener =
+    object : CashAppPayListener {
+        override fun cashAppPayStateDidChange(newState: CashAppPayState) {
+            Log.d(tag, "cashAppPayStateDidChange: ${newState::class.java}")
+            when (newState) {
+                is Approved -> {
+                    newState.responseData.apply {
+                        grants?.get(0)?.let { grant: Grant ->
+                            confirmCheckoutWithAfterpay(
+                                grantId = grant.id,
+                                customerId = grant.customerId,
+                            )
+                        }
+                    }
+                }
+
+                Declined -> {
+                    // TODO 
+                }
+               ... // other states 
+            }
+        }
+    }
+```
+
+# Step 10: Pass grant back to Afterpay
+
+
+By combining the `CheckoutV3CashAppPay` you stored before, with the newly received `grantId` and `customerId`, you can now confirm checkout with Afterpay
+
+
+```kotlin
+fun confirmCheckoutWithAfterpay(
+    grantId: String,
+    customerId: String,
+) {
+    checkoutV3CashAppPay?.let { it ->
+        Afterpay.confirmCheckoutV3WithCashAppPay(
+            grantId = grantId,
+            customerId = customerId,
+            token = it.token,
+            singleUseCardToken = it.singleUseCardToken,
+            jwt = it.jwt,
+            configuration = createCheckoutV3Configuration(),
+        ).let { result: Result<CheckoutV3Data> ->
+            result.onSuccess {
+             // TODO next step
+            }
+
+            result.onFailure {
+             // TODO
+            }
+        }
+    }
+}
+```
+
+## Step 11: Receive one-time use card details and pass back to your server for processing
```

Comment:
We don't need to send details back to server

Reply:
they need to do *something* with those card details right?